### PR TITLE
Janitorial: Refactor primary-domain component

### PR DESCRIPTION
**Pull request by aidvu**
- ES6ify
- Use redux actions for tracking

### CODE REVIEW

File: client/my-sites/domains/domain-management/primary-domain/index.jsx [hunk 1]
```diff
@@ -124,70 +128,103 @@ const PrimaryDomain = React.createClass( {
...
-	setPrimaryDomain: upgradesActions.setPrimaryDomain
-};
+export default connect(
+	state => {
+		const selectedSite = getSelectedSite( state );
+		const domains = getDomainsBySite( state, selectedSite );
 
-export default connect( mapStateToProps, mapDispatchToProps )( PrimaryDomain );
+		return {
+			domains: {
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/primary-domain/index.jsx]: An idea for a selector - `getSelectedDomain( selectedDomainName )`. No need to fetch the list and pass it as props (if the list changes, we'd re-render unnecessarily, etc.). Only `selectedDomainName` would be needed - the rest would be taken from the `state`, right? (heck, we could probably store that in the state too ;))

**Comment by aidvu** [comment]
@klimeryk you're saying this one's not used anymore? :laughing:

**Comment by klimeryk** [comment]
@mikeshelton1503, do you remember https://github.com/Automattic/wp-calypso/pull/3931 and what was the reason we left two flows for setting primary domain? IIRC, it was to check if the new one was popular and worked properly, but I think we should know that by now ;)

File: client/my-sites/domains/domain-management/primary-domain/index.jsx (hunk 1 above)
Comment (aidvu) [inline client/my-sites/domains/domain-management/primary-domain/index.jsx]: Let's see if we remove it. If so, can revisit. :)

**Comment by mikeshelton1503** [comment]
I don't remember specifically what was discussed back then, but if we can peak at usage the button on the domain settings page vs. usage in the domains list since that change (maybe look at the last 30 days) and confirm that most of the usage is in the domains list then I'm fine removing it.

**Comment by aidvu** [comment]
Will check today. 👍

**Pull request merged**